Index: valgrind-3.15.0/mpi/Makefile.am
===================================================================
 Patch by Samuel Thibault:
 https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=946329;msg=22
--- valgrind-3.15.0.orig/mpi/Makefile.am
+++ valgrind-3.15.0/mpi/Makefile.am
@@ -42,14 +42,14 @@ libmpiwrap_@VGCONF_ARCH_PRI@_@VGCONF_OS@
 libmpiwrap_@VGCONF_ARCH_PRI@_@VGCONF_OS@_so_CPPFLAGS = -I$(top_srcdir)/include
 libmpiwrap_@VGCONF_ARCH_PRI@_@VGCONF_OS@_so_CFLAGS   = \
 	$(CFLAGS_MPI) $(MPI_FLAG_M3264_PRI) -Wno-deprecated-declarations
-libmpiwrap_@VGCONF_ARCH_PRI@_@VGCONF_OS@_so_LDFLAGS  = $(LDFLAGS_MPI)
+libmpiwrap_@VGCONF_ARCH_PRI@_@VGCONF_OS@_so_LDADD  = $(LDFLAGS_MPI)
 endif
 if BUILD_MPIWRAP_SEC
 libmpiwrap_@VGCONF_ARCH_SEC@_@VGCONF_OS@_so_SOURCES  = libmpiwrap.c
 libmpiwrap_@VGCONF_ARCH_SEC@_@VGCONF_OS@_so_CPPFLAGS = -I$(top_srcdir)/include
 libmpiwrap_@VGCONF_ARCH_SEC@_@VGCONF_OS@_so_CFLAGS   = \
 	$(CFLAGS_MPI) $(MPI_FLAG_M3264_SEC) -Wno-deprecated-declarations
-libmpiwrap_@VGCONF_ARCH_SEC@_@VGCONF_OS@_so_LDFLAGS  = $(LDFLAGS_MPI)
+libmpiwrap_@VGCONF_ARCH_SEC@_@VGCONF_OS@_so_LDADD  = $(LDFLAGS_MPI)
 endif
 
 #----------------------------------------------------------------------------
